openeo/rest/datacube.py

```diff
@@ -523,34 +523,48 @@ def reduce_tiles_over_time(self, code: str, runtime="Python", version="latest"):
         """
         return self.reduce_temporal_udf(code=code, runtime=runtime, version=version)
 
-    def apply(self, process: str, data_argument='x') -> 'DataCube':
+    def apply(self, process: Union[PGNode, str], data_argument='x') -> 'DataCube':
         # TODO #125 allow more complex sub-process-graphs?
+
+        arguments = {"data": self._pg}
+
+        if self._api_version.at_least("1.0.0"):
+            if isinstance(process, str):
+                arguments["process"] = {"process_graph": PGNode(
+                        process_id=process,
+                        arguments={data_argument: {"from_parameter": "x"}})}
+            else:
+                arguments["process"] = {"process_graph": process}
+        else:
+            if isinstance(process, str):
+                arguments["process_graph"] = PGNode(
+                        process_id=process,
+                        arguments={data_argument: {"from_parameter": "x"}})
+            else:
+                arguments["process_graph"] = process
+
         return self.process_with_node(PGNode(
             process_id='apply',
-            arguments={
-                "data": self._pg,
-                "process": {"process_graph": PGNode(
-                    process_id=process,
-                    arguments={data_argument: {"from_parameter": "x"}}
-                )},
+            arguments=arguments
                 # TODO #125 context
-            }
         ))
 
-    def reduce_temporal_simple(self, process_id="max") -> 'DataCube':
+    def reduce_temporal_simple(self, process_id="max", dim_abbr="temporal") -> 'DataCube':
```

Comment:
I would use `dimension` instead of `dim_abbr`,  as used in other places in DataCube methods

also, picking the correct dimension should be done automatically instead of manually by user: see #116. Can you put a comment `# TODO #116 determine dimension based on datacube metadata` here as well?